### PR TITLE
Implement safe arithmetic for blockchain mod

### PR DESCRIPTION
**Pull request by vlopes11**
Addresses `blockchain.rs` for issue #170

### CODE REVIEW

File: src/interpreter/blockchain.rs [hunk 1]
```diff
@@ -30,66 +31,81 @@ where
...
+            || length > MEM_MAX_ACCESS_SIZE as usize
+            || length > self.params.contract_max_size as usize
         {
             return Err(PanicReason::MemoryOverflow.into());
         }
 
-        if ssp != sp {
-            return Err(PanicReason::ExpectedUnallocatedStack.into());
-        }
+        // compiler will optmize to memset
```
Comment (mitch-fuel) [inline src/interpreter/blockchain.rs:57]: optimize

Comment (vlopes11) [reply src/interpreter/blockchain.rs]: https://github.com/FuelLabs/fuel-vm/pull/195/commits/d559692b84e3bb6c1840a67806c71b2e92cea9d9

File: src/error.rs [hunk 2]
```diff
@@ -226,6 +226,9 @@ pub enum BugId {
     ID004,
     ID005,
     ID006,
+    ID007,
+    ID008,
+    ID009,
```
Comment (adlerjohn) [inline src/error.rs:231]: Why are you adding 3 IDs here but only 2 errors further down?

Comment (vlopes11) [reply src/error.rs:231]: This ID is a fallback for when the debug symbols are not available for the backtrace. They map to unique locations of the source, and we have 3 new instances:

https://github.com/FuelLabs/fuel-vm/blob/d559692b84e3bb6c1840a67806c71b2e92cea9d9/src/interpreter/blockchain.rs#L93-L99

https://github.com/FuelLabs/fuel-vm/blob/d559692b84e3bb6c1840a67806c71b2e92cea9d9/src/interpreter/blockchain.rs#L102-L105

https://github.com/FuelLabs/fuel-vm/blob/d559692b84e3bb6c1840a67806c71b2e92cea9d9/src/interpreter/blockchain.rs#L420-L422

Comment (vlopes11) [reply src/error.rs:231]: We are being over-cautious now, even with checks that logically can't panic. The idea is to reduce these checks when we enter the optimization phase

**Review by adlerjohn — COMMENTED** [review]
looks fine to me, but will defer to client team

**Review by mitch-fuel — APPROVED** [review]
I have tested all of these checks in my associated PR, approved

**Pull request merged**